pkg/util/fxutil/provide_comp.go

```diff
@@ -293,13 +294,23 @@ func coerceStructTo(input reflect.Value, outType reflect.Type, oldEmbed, newEmbe
 	return result
 }
 
+// FxPrerequisites returns all of our adapters from compdef types to fx types
+func FxPrerequisites() fx.Option {
```

Comment:
What about renaming to something that feels more custom to the Agent? 

`FxAgentOptions` or `AgentFxOptions` or `CustomFxAgentOptions`

What do you think?

Reply:
Hmmm, that's a fine suggestion. These `fx.Option`s being added are necessary for a lot of the fxutil wrappers, they assume that certain types are being provided, and I would like to capture that requirement in the name. How about `FxAgentBase()`?